### DISCUSSION TITLE
Drop Browser support older than January 2021 in 2.10 ?

### DISCUSSION THREAD

**Discussion by lubber-de** [fomantic/Fomantic-UI#2854] [Polls]
You people [already decided to drop IE11 by 2.10](https://github.com/fomantic/Fomantic-UI/discussions/2613). Great!
Now we also think about going a step further and also make use of modern CSS Selectors like [:where](https://caniuse.com/?search=%3Awhere) or [:is](https://caniuse.com/?search=%3Ais) 

Those techniques are at least supported by all modern Browsers released in January 2021 (Chrome/Edge 88, Safari 14, Firefox 78)

This step would allow to reduce the overall CSS but won't work in older Browsers released < 2021 and to also drop support for iPhone < 7 (older than 2015), because [only iPhone 7+ can install iOS 14+](https://iosref.com/ios)

**Comment by designosis** [comment]
My first thought was to wait for **3.0** ... why risk alienating an unknown % of visitors? Then I remembered that the interweb (and sources such as [this](https://www.w3schools.com/browsers/)) could help me estimate that %:

- Chrome/Edge < 88 in July 2023: **0.05%**
- Safari < 14 in July 2023: **0.08%**
- Firefox < 78 in July 2023: **0.01%**
- One [source](https://www.statista.com/statistics/565270/apple-devices-ios-version-share-worldwide/) says iOS < 14 accounts for **4%** of all iOS users in in February 2023 ... but another [source](https://telemetrydeck.com/blog/ios16-market-share-02-23/#:~:text=As%20of%20February%201%2C%202023,iOS%2014%3A%200.2%25) has that same number at only **0.5%**

If 61%[*](https://explodingtopics.com/blog/mobile-internet-traffic) of web browsing happens on mobile, and roughly 50%[*](https://daringfireball.net/2023/05/browser_and_os_market_share_trends_in_the_mobile_age) of the world uses iPhones, that high estimate of 4% becomes **1.2%**.

On the one hand, weird datasets are ubiquitous, and I have low confidence in these numbers. On the other hand, even with the possibility of losing 1% of the userbase, I'd rather have things updated in **2.10.0**. Why? Peak performance for the many, and natural selection for the luddites.

**Comment by aggregate1166877** [comment]
I'll add my 2 cents with a disclaimer: I don't use Semantic or Fomantic in legacy projects, so my example isn't necessarily applicable to me personally. But I do use other modern libraries in legacy projects.

My own use-case: Not all of us target browsers. For example, I target Android 4.3 to Android 9 via an old project called Crosswalk. It allows me to run modern libs (such as Three.js 130+) on phones from 2011 (such as the Samsung Galaxy S2). Crosswalk uses a custom webview based on Chromium version 52 (because native webviews weren't stable back then [hell, they're barely stable now]). Because Chromium was fast at v52 already, my game prototypes reach 55 frames per second on said 2011 S2, which appears to be the device's limit. And security isn't a concern because the apps speak only to my own servers. Any library that dropped support for old tech would force me to downgrade the library version, choose another library, or write my own from scratch.

Would I use S/Fomantic on an ancient device? ~~Probably not~~ (actually I'm working on a chat app right now that will most likely be back-ported, but that uses Semantic anyway). Nonetheless, I thought it might shed some insight as to why old tech is relevant in 2024.

**Reply by lubber-de** [reply]
Decision made: by #3528 we will drop every browser older than 12/2022
